gemrb/GUIScripts/iwd/CharGen.py

```diff
@@ -1535,7 +1535,10 @@ def SkillsPress():
 		elif DruidSpell!="*":
 			Skill = GemRB.LoadTable("SKILLRNG").GetValue(str(Level), "STEALTH")
 			GemRB.SetPlayerStat (MyChar, IE_STEALTH, Skill)
-			RacialEnemySelect()
+			if ClassName == "RANGER":
+				RacialEnemySelect()
+			else:
+			    SkillsSelect()
```

Comment:
Come on, look at your diffs. A better way to check who should get it is by checking if the HATERACE column of clsskills.2da (CommonTables.ClassSkills) has an entry.

Reply:
in addition to what you said, this probably shouldn't be under a check for `DruidSpell` either. might technically cover all built in classes, but I assume mods can add this to anything they want.

Reply:
I always look at diff before committing, but as it turns out, diff-so-fancy that I use to make it more readable, shows tabs as four spaces, that's why I couldn't see it. Changed shiftwidth to 8 spaces, so it will be now readable in editor. If I may express my opinion, then let me say that python code style and formatting is rather peculiar. It would be easier to maintain consistent style if code formatter was being used.

> A better way to check who should get it is by checking if the HATERACE column of clsskills.2da (CommonTables.ClassSkills) has an entry.

I see. Is there more comprehensive description of game files than IESDP? It has some useful knowledge, but it's not describing, for example, `clsskills.2da` and its fields.

Reply:
Nothing in gemrb/unhardcoded/ is on iesdp, since they are all our files. There is gemrb/docs/en/Tables/, but we'll eventually move them inside the table files themselves, since we otherwise don't remember to update the docs ever.

Reply:
> in addition to what you said, this probably shouldn't be under a check for DruidSpell either. might technically cover all built in classes, but I assume mods can add this to anything they want.

That's good idea I guess, but will complicate checks as race hate opens later skills select, so we will have to ensure it is not opened twice.

Reply:
I wouldn't bother with this at all. We rely on only rangers having this table in core as well. It's the druid that doesn't get any special skills, so you can just replace the `DruidSpell!="*"` check.

Reply:
```diff
diff --git a/gemrb/GUIScripts/iwd/CharGen.py b/gemrb/GUIScripts/iwd/CharGen.py
index 9962d799e1..fe6bf4c543 100644
--- a/gemrb/GUIScripts/iwd/CharGen.py
+++ b/gemrb/GUIScripts/iwd/CharGen.py
@@ -1526,6 +1526,7 @@ def SkillsPress():
 	MageSpell = CommonTables.ClassSkills.GetValue (ClassName, "MAGESPELL")
 	IsBard = CommonTables.ClassSkills.GetValue (ClassName, "BARDSKILL")
 	IsThief = CommonTables.ClassSkills.GetValue (ClassName, "THIEFSKILL")
+	HateRace = CommonTables.ClassSkills.GetValue(ClassName, "HATERACE")
 	LUSkillsSelection.SkillsNullify (MyChar)
 
 	if SkillsState == 0:
@@ -1535,10 +1536,6 @@ def SkillsPress():
 		elif DruidSpell!="*":
 			Skill = GemRB.LoadTable("SKILLRNG").GetValue(str(Level), "STEALTH")
 			GemRB.SetPlayerStat (MyChar, IE_STEALTH, Skill)
-			if ClassName == "RANGER":
-				RacialEnemySelect()
-			else:
-			    SkillsSelect()
 		elif IsBard!="*":
 			Skill = GemRB.LoadTable(IsBard).GetValue(str(Level), "PICK_POCKETS")
 			GemRB.SetPlayerStat (MyChar, IE_PICKPOCKET, Skill)
@@ -1546,6 +1543,12 @@ def SkillsPress():
 		else:
 			SkillsState = 1
 
+		if HateRace != "*":
+			RacialEnemySelect()
+		elif IsThief!="*" or DruidSpell!="*":
+			SkillsSelect()
+
+
 	if SkillsState == 1:
 		ProficienciesSelect()
 ```

Something like this seems to work, but give me more time to test it further.

Reply:
Is there some other reason druids need a call to SkillsSelect?

Reply:
Other reason? I don't think I'm on par with you on this, it's just needed distribute skill points. `RacialEnemySelect()` implies calling `SkillsSelect()` afterwards. If you remove `or DruidSpell!="*", then simply skills window doesn't open anymore for druids.

Reply:
Yeah, but what skills do they have that they need the window? None that I'm aware ...

Reply:
Oh, I mistook skills for proficiencies. So, to rephrase myself, proficiencies window doesn't open. Perhaps `SkillSelect()` sets `SkillsState = 1` or something like that. I'll check that in the evening.

Reply:
Just look at the whole if-block from the diff — if nothing matches SkillsState will be set to 1 and proficiencies immediately opened.

Reply:
I've seen that, that's why I thought it might be solution, but it could be that `SkillsSet()` manipulates global state, and sets something druid might need. After quick check, it appears that setting SkillsState is sufficient.